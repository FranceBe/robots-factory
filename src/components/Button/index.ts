import ButtonDefault, { Button } from 'components/Button/Button.component'

export { Button }

export default ButtonDefault
