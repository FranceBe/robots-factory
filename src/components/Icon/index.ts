import { FAIconByIconProps, Icon } from 'components/Icon/Icon.component'

export { FAIconByIconProps, Icon }
