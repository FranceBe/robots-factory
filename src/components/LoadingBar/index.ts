import { LoadingBar } from 'components/LoadingBar/LoadingBar.component'

export { LoadingBar }
