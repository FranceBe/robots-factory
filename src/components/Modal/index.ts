import ModalDefault, { Modal } from 'components/Modal/Modal.component'

export { Modal }

export default ModalDefault
