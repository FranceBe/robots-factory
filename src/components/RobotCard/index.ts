import { RobotCard } from 'components/RobotCard/RobotCard.component'

export { RobotCard }
