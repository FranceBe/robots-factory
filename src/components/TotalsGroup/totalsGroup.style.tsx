import styled from 'styled-components'
import { spaces } from 'styles/variables'

export const CardContainer = styled.div`
  width: 90%;
  .totals-component:not(:nth-last-child(2)) {
    margin-bottom: ${spaces.small};
  }
  .totals-component:nth-last-child(2) {
    margin-bottom: ${spaces.regular};
  }
`
