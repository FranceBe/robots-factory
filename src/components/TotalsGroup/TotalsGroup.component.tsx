// Totals group component is a list of all the totals needed for the games
import { Icon } from 'components/Icon'
import { Totals } from 'components/Totals'
import { CardContainer } from 'components/TotalsGroup/totalsGroup.style'
import { ResourceType } from 'contexts/robotsContext/robotContext'
import React from 'react'
import { IconEnum, ThemeType } from 'utils/common.enum'

export const TotalsGroup: React.FC<{ totals: ResourceType }> = ({ totals }) => {
  return (
    <CardContainer data-testid={'totals-group'}>
      <Totals label={'Foo'} total={totals.foo} type={ThemeType.primary} />
      <Totals label={'Bar'} total={totals.bar} type={ThemeType.primary} />
      <Totals label={'Foobar'} total={totals.foobar} type={ThemeType.primary} />
      <Totals
        label={<Icon type={IconEnum.robot} />}
        total={totals.robot}
        type={ThemeType.secondary}
      />
    </CardContainer>
  )
}

export default TotalsGroup
