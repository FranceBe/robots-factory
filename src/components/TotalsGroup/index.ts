import { TotalsGroup } from 'components/TotalsGroup/TotalsGroup.component'

export { TotalsGroup }
