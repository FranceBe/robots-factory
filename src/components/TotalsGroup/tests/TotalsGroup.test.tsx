import '@testing-library/jest-dom/extend-expect'

import { render, screen } from '@testing-library/react'
import { TotalsGroup } from 'components/TotalsGroup'
import { ResourceType } from 'contexts/robotsContext/robotContext'
import React from 'react'

describe('TotalsGroup component', () => {
  const totals: ResourceType = {
    bar: 0,
    foo: 4,
    foobar: 10,
    robot: 6,
  }

  const initTest = () => render(<TotalsGroup totals={totals} />)

  it('should match snapshot', () => {
    const { container } = initTest()

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render 4 Totals component', () => {
    initTest()

    expect(screen.getAllByTestId('totals-component')).toHaveLength(4)
  })
  it('should render each labels and total as text', () => {
    initTest()

    expect(screen.getByText('Foo')).toBeInTheDocument()
    expect(screen.getByText('4')).toBeInTheDocument()
    expect(screen.getByText('Bar')).toBeInTheDocument()
    expect(screen.getByText('0')).toBeInTheDocument()
    expect(screen.getByText('Foobar')).toBeInTheDocument()
    expect(screen.getByText('10')).toBeInTheDocument()
    expect(screen.getByRole('img', { hidden: true })).toBeInTheDocument()
    expect(screen.getByRole('img', { hidden: true })).toHaveClass('fa-robot')
    expect(screen.getByText('6')).toBeInTheDocument()
  })
})
