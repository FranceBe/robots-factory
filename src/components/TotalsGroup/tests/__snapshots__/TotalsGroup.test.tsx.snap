// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalsGroup component should match snapshot 1`] = `
<div
  class="sc-gtsrHT cIIqls"
  data-testid="totals-group"
>
  <div
    class="sc-bdnxRM kQbngr totals-component"
    data-testid="totals-component"
    type="primary"
  >
    <div>
      Foo
    </div>
    <div>
      4
    </div>
  </div>
  <div
    class="sc-bdnxRM kQbngr totals-component"
    data-testid="totals-component"
    type="primary"
  >
    <div>
      Bar
    </div>
    <div>
      0
    </div>
  </div>
  <div
    class="sc-bdnxRM kQbngr totals-component"
    data-testid="totals-component"
    type="primary"
  >
    <div>
      Foobar
    </div>
    <div>
      10
    </div>
  </div>
  <div
    class="sc-bdnxRM jceLen totals-component"
    data-testid="totals-component"
    type="secondary"
  >
    <div>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-robot fa-w-20 "
        data-icon="robot"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32,224H64V416H32A31.96166,31.96166,0,0,1,0,384V256A31.96166,31.96166,0,0,1,32,224Zm512-48V448a64.06328,64.06328,0,0,1-64,64H160a64.06328,64.06328,0,0,1-64-64V176a79.974,79.974,0,0,1,80-80H288V32a32,32,0,0,1,64,0V96H464A79.974,79.974,0,0,1,544,176ZM264,256a40,40,0,1,0-40,40A39.997,39.997,0,0,0,264,256Zm-8,128H192v32h64Zm96,0H288v32h64ZM456,256a40,40,0,1,0-40,40A39.997,39.997,0,0,0,456,256Zm-8,128H384v32h64ZM640,256V384a31.96166,31.96166,0,0,1-32,32H576V224h32A31.96166,31.96166,0,0,1,640,256Z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div>
      6
    </div>
  </div>
</div>
`;
